### PR TITLE
IDEMPIERE-5468 - Negate option in parameter panel when component is a Chosen Multiple Selection

### PR DESCRIPTION
**Pull request by matheus-marcelinux**
https://idempiere.atlassian.net/browse/IDEMPIERE-5468

# Pull Request Checklist

## Checklist:

- [X] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [X] My code follows the best practices of this project
- [X] I have performed a self-review of my own code
- [X] My code is easy to understand and review. 
- [X] In hard-to-understand areas, I have commented my code.
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have made corresponding changes to the documentation as follows:
- - [x] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [x] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/db/AdempiereDatabase.java [hunk 1]
```diff
@@ -344,9 +344,10 @@ public Connection getDriverConnection (String dbUrl, String dbUid, String dbPwd)
 	/**
 	 * @param columnName
 	 * @param csv comma separated value
+	 * @param isNotClause
 	 * @return subset sql clause
 	 */
-	public String intersectClauseForCSV(String columnName, String csv);
+	public String intersectClauseForCSV(String columnName, String csv, boolean isNotClause);
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/db/AdempiereDatabase.java:350]: for backward compatibility, please also keep the existing method signature.

**Review by hengsin — CHANGES_REQUESTED** [review]
please see comments at ticket too, thanks.

File: org.adempiere.base/src/org/compiere/db/AdempiereDatabase.java (hunk 1 above)
Comment (matheus-marcelinux) [reply org.adempiere.base/src/org/compiere/db/AdempiereDatabase.java:357]: done

**Comment by hengsin** [comment]
hi @matheus-marcelinux , please fix merge conflict.

**Comment by hengsin** [comment]
[pr1567.patch.txt](https://github.com/idempiere/idempiere/files/10112484/pr1567.patch.txt)
- icon and UI changes, see screenshot posted to ticket.
- added **z-icon-IncludeSelected** and **z-icon-ExcludeSelected** for the 2 state indicator. This make it it easier for third party theme to change it.
- added setting of sclass **btn-negate-exclude** and **btn-negate-include**.
- remove the use of **btn-negate-gray** and  **btn-negate-red css** class.

Note:
- please change idempiere.theme.default version at lang-addon.xml to resolve conflict
- translate **Exclude selected values** and **Include selected values**, i.e create ad_message record for both and add the corresponding migration script to this PR.

**Review by hengsin — APPROVED** [review]
tested

**Comment by CarlosRuiz-globalqss** [comment]
The migration script 202211301704_IDEMPIERE-5468.sql is wrong, it doesn’t have centralized IDs.

**Pull request merged**